Tweak the loading indicator so its less jarring when changing views

nwmac opened the pull request:
Most of the time when you navigate around the UI, a loading indicator is shown on page changes as data is fetched - this currently overlays the entire browser view, which is quite jarring.

This PR changes this behaviour - page views will now only show the loading indicator over the content area (not the side nav or header).

I've added a mode to the Loading component to allow configurability of what portion of the window should be covered for the indicator.

The Global Loading component uses the 'full' mode - so when you log in, the Loading indicator still appears over the entire browser view.

I think this overall gives a better user experience.

lvuch reviewed (commented):
I pushed a small change to loading.vue to use principal translation. Visually this feels tidy.

westlywright reviewed (approved):
I like this! I pulled it down and even though the change is subtle I immediately felt like the UI was snappier.

vincent99 commented:
I don't get it...  the result is that the entire page is still blocked from interaction (because the overlay DIV still covers the whole page) but only some of the page _looks like it_ is blocked (because the bg is now on the content DIV instead of the overlay).  So now there's things that look clickable, but aren't.

And some routes end up triggering both kinds in succession, so you see the background flash and the Loading text move during the transition.

These are more obvious if you set Chrome to simulate a slow connection.

(BTW the `Loading...` text was meant to be a placeholder for a graphical thing like Ember has, but just hasn't been done...)

![loading](https://user-images.githubusercontent.com/753917/107676842-d1eeaf80-6c56-11eb-9a65-bcf337279ecd.gif)

vincent99 commented on components/Loading.vue:
```diff
@@ -22,7 +22,7 @@ export default {
     <div class="overlay"></div>
     <div class="content" :class="{ 'content-content-mode' : mode === 'content', 'content-main-mode' : mode === 'main' }">
-      Loading...
+      <t k="principal.loading" />
```
This should get a separate translation entry, `principal.loading` is used when a principal (small user icon + their name box like in add members to things) is being loaded inline on the page and you probably wouldn't expect changing that to affect every page or vice versa.

It's not particularly important for this usage, but `&hellip;` produces a proper "…" vs "..."; they are one symbol instead of 3 and slightly closer together.  You just need the translation to be output raw without html escaping to use it since it's an entity.

nwmac replied:
Lauren made that change, I am agree, it should be separate one.

nwmac commented:
@vincent99 I assumed there was a good reason we prevented nav when the loading indicator was shown - sounds like there is not, so I will update, allowing users to access the nav while it it shown (in the case its not global).

I'll look at the issue where you get one global loading shortly followed by the second loading to see what can be done.

vincent99 commented:
> @vincent99 I assumed there was a good reason we prevented nav when the loading indicator was shown - sounds like there is not, so I will update, allowing users to access the nav while it it shown (in the case its not global).

You're right, I do expect there probably _will_ be problems that are going to come up with allowing navigation in the middle of another transition.  But ran into it being visually available but not actually clickable before figuring out what those might be.  My first try was going to be changing the selected cluster.

nwmac commented:
@vincent99 Okay - updated as per other comments - the issue with the two indicators is caused by the lazy loading of code (I think) when you navigate to some pages and then it loads data giving the second loading indicator afterwards. I've added a delay before showing the indicator which alleviates this in the fast connection case, where this is more jarring.

Fixed the translation and removed the overlay across the whole page that prevented nav.

nwmac commented:
Reviewed with Richard - I've addressed the feedback from Vince.
